camelot/view/validator.py

```diff
@@ -82,51 +82,29 @@ def validate(self, input_, pos):
             return (QtGui.QValidator.State.Intermediate, input_, pos)
         return (QtGui.QValidator.State.Acceptable, input_, pos)
 
-class RegexReplaceValidator(QtGui.QValidator, AbstractValidator):
-
-    @dataclass    
-    class State(DataclassSerializable):
-
-        regex: str = None
-        regex_repl: str = None
-        example: str = None
-        deletechars: str = ' -./#,'
-
-    def __init__(self, parent=None):
-        super().__init__(parent)
-        self.state = self.State()
-
-    def set_state(self, state):
-        state = state or dict()
-        assert isinstance(state, dict)
-        self.state = self.State(**state)
-        if (regex := self.state.regex) is not None:
-            self.state.regex = re.compile(regex)
+@dataclass
+class RegexReplaceValidatorState(DataclassSerializable):
 
-    def validate(self, qtext, position):
-        ptext = str(qtext).upper()
-        if not ptext:
-            return (QtGui.QValidator.State.Acceptable, qtext, 0)
-
-        validity = self.validity(ptext)
-        if validity.valid:
-            return (QtGui.QValidator.State.Acceptable, validity.formatted_value, len(validity.formatted_value or ''))
-        return (QtGui.QValidator.State.Intermediate, qtext, position)
+    regex: str = None
+    regex_repl: str = None
+    example: str = None
+    deletechars: str = ' -./#,'
 
     def validity(self, value):
```

Comment:
zouden we van 'validity' geen 'classmethod' maken, met de 'state' als argument, waarbij state de geserialiseerde state is ?